third_party/address_sorting/address_sorting.c

```diff
@@ -255,7 +255,7 @@ static int compare_source_dest_labels_match(
     second_label_matches = 1;
   }
   if (first_label_matches != second_label_matches) {
-    return first_label_matches ? 1 : 1;
+    return first_label_matches ? -1 : 1;
```

Comment:
I'm not sure what this function is trying to do. Can you briefly explain what source_addr, dest_addr is? 
It would also be nice to explain what the bug was.

This certainly looks like a small typo in the original code. Is that all this is?

Also first_label_matches and second_label_matches should be boolean..